images/virtualization-artifact/pkg/controller/vmiplease/internal/deletion_handler.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/deckhouse/virtualization-controller/pkg/controller/service"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/vmiplease/internal/state"
+	"github.com/deckhouse/virtualization/api/core/v1alpha2/vmipcondition"
+	"github.com/deckhouse/virtualization/api/core/v1alpha2/vmiplcondition"
+)
+
+const DeletionHandlerName = "DeletionHandler"
+
+type DeletionHandler struct {
+	client                                  client.Client
+	logger                                  logr.Logger
+	virtualMachineIPLeasesRetentionDuration string
```

Comment:
It can be named just retentionDuration, this struct is in package specific to virtualMachineIPLeases